### DISCUSSION TITLE
.wave in css to slint

### DISCUSSION THREAD

**Discussion by sundaritisthatis** [slint-ui/slint#4172] [Q&A]
Need example to extend the gradient to wave feature in slint using colors - kindly help rendering it

**Comment by ghost** [comment]
Hi. Do you request to have such a feature for the slint brush or do you want help to create an wave right now in Slint? What is always possible for custom drawing is to draw something using an image in Slint.

**Comment by sundaritisthatis** [comment]
https://github.com/ecomfe/echarts-liquidfill

am trying a similar one in Slint - able to create gradient with svg commands but animation of it is a struggle as property with string type doesn't hold for animation

**Comment by ghost** [comment]
I think based way to implement that in Slint is to use animations and Path https://slint.dev/releases/1.3.0/docs/slint/src/language/builtins/elements#path.

**Comment by ogoffart** [comment]
I've tried to implement that.  It's a bit naive but it works: [Link](https://slintpad.com/?snippet=import+%7B+AboutSlint%2C+Button%2C+VerticalBox+%7D+from+%22std-widgets.slint%22%3B%0A%0Aexport+component+Wave+%7B%0A++++in+property+%3Cbrush%3E+fill%3A+blue%3B%0A++++in+property+%3Cangle%3E+a%3A+1turn+*+animation-tick%28%29%2F3s+%3B%0A++++in+property+%3Cfloat%3E+p%3A+1%3B+%0A++++in+property+%3Cfloat%3E+w%3A+0.1%3B%0A++++in+property+%3Cangle%3E+f%3A+1turn+%3B%0A++++in+property+%3Cfloat%3E+h%3A+0.5%3B%0A++++preferred-width%3A+100%25%3B%0A++++preferred-height%3A+100%25%3B%0A%0A++++Path+%7B%0A++++++++width%3A+100%25%3B%0A++++++++height%3A+100%25%3B%0A++++++++viewbox-height%3A+1%3B%0A++++++++viewbox-width%3A+1%3B%0A++++++++fill%3A+fill%3B%0A++++++++MoveTo+%7B+x%3A+0%3B++++++y%3A+h+%2B+w+*+cos%28a%29%3B+%7D%0A++++++++CubicTo+%7B+x%3A++1%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B++1%2F16+*+f%29%3B+control-1-x%3A++1%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%28+1%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A++1%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%28+1%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A++2%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B++2%2F16+*+f%29%3B+control-1-x%3A++2%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%28+2%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A++2%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%28+2%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A++3%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B++3%2F16+*+f%29%3B+control-1-x%3A++3%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%28+3%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A++3%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%28+3%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A++4%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B++4%2F16+*+f%29%3B+control-1-x%3A++4%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%28+4%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A++4%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%28+4%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A++5%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B++5%2F16+*+f%29%3B+control-1-x%3A++5%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%28+5%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A++5%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%28+5%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A++6%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B++6%2F16+*+f%29%3B+control-1-x%3A++6%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%28+6%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A++6%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%28+6%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A++7%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B++7%2F16+*+f%29%3B+control-1-x%3A++7%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%28+7%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A++7%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%28+7%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A++8%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B++8%2F16+*+f%29%3B+control-1-x%3A++8%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%28+8%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A++8%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%28+8%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A++9%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B++9%2F16+*+f%29%3B+control-1-x%3A++9%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%28+9%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A++9%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%28+9%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A+10%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B+10%2F16+*+f%29%3B+control-1-x%3A+10%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%2810%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A+10%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%2810%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A+11%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B+11%2F16+*+f%29%3B+control-1-x%3A+11%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%2811%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A+11%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%2811%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A+12%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B+12%2F16+*+f%29%3B+control-1-x%3A+12%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%2812%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A+12%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%2812%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A+13%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B+13%2F16+*+f%29%3B+control-1-x%3A+13%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%2813%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A+13%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%2813%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A+14%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B+14%2F16+*+f%29%3B+control-1-x%3A+14%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%2814%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A+14%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%2814%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A+15%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B+15%2F16+*+f%29%3B+control-1-x%3A+15%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%2815%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A+15%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%2815%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++CubicTo+%7B+x%3A+16%2F16%3B+y%3A+h+%2B+w+*+cos%28a+%2B+16%2F16+*+f%29%3B+control-1-x%3A+16%2F16+-+1%2F32%3B+control-1-y%3A+h+%2B+w+*+cos%28a+%2B+%2816%2F16+-+1%2F32%29+*+f%29%3B+control-2-x%3A+16%2F16+-+1%2F32%3B+control-2-y%3A+h+%2B+w+*+cos%28a+%2B+%2816%2F16+-+1%2F32%29+*+f%29%3B+%7D+%0A++++++++LineTo+%7B%0A++++++++++++x%3A+1%3B%0A++++++++++++y%3A+1%3B%0A++++++++%7D%0A++++++++LineTo+%7B%0A++++++++++++x%3A+0%3B%0A++++++++++++y%3A+1%3B%0A++++++++%7D%0A++++++++LineTo+%7B%0A++++++++++++x%3A+0%3B%0A++++++++++++y%3A+0.5%3B%0A++++++++%7D%0A++++%7D%0A%0A%7D%0A%0Aexport+component+Demo+%7B%0A++++Rectangle+%7B%0A++++++++border-radius%3A+self.width+%2F+2%3B%0A++++++++width%3A+min%28parent.width%2C+parent.height%29%3B%0A++++++++height%3A+self.width%3B%0A++++++++border-color%3A+darkblue%3B%0A++++++++border-width%3A+5px%3B%0A++++++++clip%3A+true%3B%0A++++++++Wave+%7B+f%3A+1turn%3B+h%3A+1+-+0.6+-+0.01%3B+fill%3A+%40linear-gradient%28180deg%2C+%230001%2C+transparent%29%3B+%7D%0A++++++++Wave+%7B+f%3A+1turn%3B+h%3A+1+-+0.6++++++%3B+fill%3A+%2300f%3B+%7D%0A%0A++++++++Wave+%7B+f%3A+1.2turn%3B+h%3A+1+-+0.5+-+0.01%3B+fill%3A+%40linear-gradient%28180deg%2C+%230001%2C+transparent%29%3B+%7D%0A++++++++Wave+%7B+f%3A+1.2turn%3B+h%3A+1+-+0.5++++++%3B+fill%3A+%230084ff%3B+%7D%0A++++++++%0A++++++++Wave+%7B+w%3A+0.05%3B+f%3A+1.5turn%3B+h%3A+1+-+0.4+-+0.01%3B+fill%3A+%40linear-gradient%28180deg%2C+%230001%2C+transparent%29%3B+%7D%0A++++++++Wave+%7B+w%3A+0.05%3B+f%3A+1.5turn%3B+h%3A+1+-+0.4++++++%3B+fill%3A+%2300ccff%3B+%7D%0A++++++++%0A++++%7D%0A++++%0A%0A%7D%0A)

**Comment by sundaritisthatis** [comment]
been curious - you are awesome - thanks - will definitely be with Slint team for such a idea to contribution

**Comment by sundaritisthatis** [comment]
@ogoffart can i get any resource to learn these things both in slint and ui things
